openedx/core/djangoapps/content/block_structure/config/__init__.py

```diff
@@ -10,12 +10,51 @@
 from .models import BlockStructureConfiguration
 
 # Switches
+# .. toggle_name: block_structure.invalidate_cache_on_publish
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: When enabled, the block structure cache is invalidated when changes to
+#   courses are published. Enabling this toggle is useful for non-expiring block structure cache.
+#   It is also useful to enable this when `block_structure.storage_backing_for_cache` is enabled
+#   because it guarantees the storage, containing the storage-backed structures, is updated with
+#   the latest changes.
+# .. toggle_use_cases: opt_in
+# .. toggle_creation_date: 2017-02-23
+# .. toggle_tickets: https://github.com/edx/edx-platform/pull/14571
```

Comment:
@nizarmah Not a blocker for this PR, but:

In order to further reduce the complexity in the `block_structure` code. I believe we can DEPR this particular toggle as well.  This toggle also remains disabled in our own production environment.  It was created as a toggle while the team was determining how best to stabilize the `block_structure` framework in production.

The default is to NOT invalidate the cache immediately. That is, to rely on eventual consistency. Some clients may still continue to use the old cached value, while the new cached value is being computed.  This is a more robust solution for a distributed environment. I would be surprised if another Open edX instance really needed this toggle enabled. We can check with the community through a DEPR ticket.

Reply:
@nasthagiri 

> The default is to NOT invalidate the cache immediately. That is, to rely on eventual consistency. Some clients may still continue to use the old cached value, while the new cached value is being computed. This is a more robust solution for a distributed environment.

I still have a lot to learn, so I'm glad you clarified that 👍🏼 

> I would be surprised if another Open edX instance really needed this toggle enabled. We can check with the community through a DEPR ticket.

I noticed it was enabled on one of our instances at OpenCraft with a claim that it helps guarantee that the storage backed block structures are updated. But after checking your comment and the related code, I went ahead and tested the changes, while storage backed block structures are enabled.

So what this does, **while storage backed block structures are enabled**, is: deletes ALL existing storage backed block structures from BlockStructureStore (storage and/or cache), and then saves the new structure.

But we already have `BLOCK_STRUCTURES_SETTINGS['PRUNING_ACTIVE']`, so I think it should be safe to deprecate that flag. But for now, I won't indicate that in my annotation, if you don't mind.

I ended up removing the lines related to why this toggle should be enabled. I substituted them with information about what happened when storage backed block structures are enabled as well.

```diff
- #   courses are published. Enabling this toggle is useful for non-expiring block structure cache.
- #   It is also useful to enable this when `block_structure.storage_backing_for_cache` is enabled
- #   because it guarantees the storage, containing the storage-backed structures, is updated with
- #   the latest changes.
+ #   courses are published. If `block_structure.storage_backing_for_cache` is active, all block
+ #   structures related to the published course are also cleared from storage.
```

Let me know please if you'd like me to make further changes 👍🏼 cc @robrap

Reply:
Sounds good.